bin/cloud-provision.sh

```diff
@@ -16,6 +16,53 @@ fi
 NOVARC_PATH=$1
 SPI_CREDENTIALS_PATH=$2
 OPENSTACK_CREDENTIALS_DIR="$JENKINS_HOME/.openstack"
+PRIVATE_KEY_PATH=~/.canonistack/${OS_USERNAME}_${OS_REGION_NAME}.key
+
+wait_for_ssh(){
+    local ip="$1"
+    local id="$2"
+    retry=30
+    while ! ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ubuntu@"$ip" true; do
+        retry=$(( retry - 1 ))
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for ssh. Aborting!"
+            openstack server delete "$id"
+            exit 1
+        fi
+        sleep 10
+    done
+}
+
+wait_for_ip(){
+    local INSTANCE_ID=$1
+    local retry=30
+    local INSTANCE_IP=""
+    INSTANCE_IP=$(openstack server show $INSTANCE_ID | grep 'addresses' | awk '{print $4}' | cut -d= -f2)
```

Comment:
You can pass -c to select a column. I was using something like this, but I'm sure you can prettify it with awk and remove the grep pipe:
ip=$(openstack server show -c addresses -f shell $INSTANCE_ID | cut -f3 -d = | tr -d '"')